education grading categories (#20764)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/educationassignment-updatesingle-gradingcategories.md

---
title: "Update gradingCategory"
description: "Update a single gradingCategory."
author: "AshwaniBansal1"
ms.localizationpriority: medium
ms.prod: "education"
doc_type: apiPageType
---

# Update gradingCategory

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Update a single gradingCategory on the [educationAssignmentSettings](../resources/educationassignmentsettings.md). Only teachers can perform this operation.

Reviewer: ```suggestion
Update a single [gradingCategory](../resources/assignment-gradingcategories.md) on the [educationAssignmentSettings](../resources/educationassignmentsettings.md). Only teachers can perform this operation.
```

---

> title: "Update gradingCategory"

Reviewer: What folder will this appear under? This one seems like it should live under Assignment Settings.

Reviewer 2: yes, it will be there, including add and remove pages.
Three pages have been added under assignments in the left side menu.

> description: "Update single gradingCategory to the assignments."

Reviewer: ```suggestion
description: "Update a single gradingCategory."
```

author: "AshwaniBansal1"
ms.localizationpriority: medium
ms.prod: "education"
doc_type: apiPageType
---

# Add gradingCategory

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Update single gradingCategory to an [educationAssignment](../resources/educationassignment.md). Only teachers can perform this operation.

Reviewer: ```suggestion
Update a single gradingCategory on the [educationAssignmentSettings](../resources/educationassignmentsettings.md). Only teachers can perform this operation.
```

|Permission type      | Permissions (from least to most privileged)              |
|:--------------------|:---------------------------------------------------------|
|Delegated (work or school account) |  EduAssignments.ReadWriteBasic, EduAssignments.ReadWrite  |
|Delegated (personal Microsoft account) |  Not supported.  |
|Application | Not supported. |

## HTTP request
<!-- { "blockType": "ignored" } -->
```http
PATCH /education/classes/{classId}/assignmentSettings/gradingCategories/{gradingCategoryId}
```

## Request headers
| Header       | Value |
|:---------------|:--------|
| Authorization  | Bearer {token}. Required.  |

## Request body

In the request body, supply the values for relevant fields that should be updated. Existing properties that are not included in the request body will maintain their previous values or be recalculated based on changes to other property values. For best performance, don't include existing values that haven't changed.

| Property     | Type        | Description |
|:-------------|:------------|:------------|
|displayName|String|The new name for a grading category.|

## Response

> If successful, this method returns a `200 OK` response code and an updated [gradingCategory](../resources/educationrubric.md) object in the response body

Reviewer: ```suggestion
If successful, this method returns a `200 OK` response code and an updated [gradingCategory](../resources/assignment-gradingcategories.md) object in the response body.
```
